ingestion/src/metadata/orm_profiler/api/workflow.py

```diff
@@ -220,34 +220,60 @@ def get_profile_query(self, entity: Table) -> Optional[str]:
 
         return None
 
-    def get_partition_details(self, entity: Table) -> Optional[TablePartitionConfig]:
+    def get_partition_details(self, entity: Table) -> Optional[PartitionProfilerConfig]:
         """Get partition details
 
         Args:
             entity: table entity
         """
-        if (
-            not hasattr(entity, "serviceType")
-            or entity.serviceType != DatabaseServiceType.BigQuery
-        ):
-            return None
-
         entity_config: Optional[TableConfig] = self.get_config_for_entity(entity)
-        if entity_config:
+
+        if entity_config:  # check if a yaml config was pass with partition definition
             return entity_config.partitionConfig
 
-        if hasattr(entity, "tablePartition") and entity.tablePartition:
-            if entity.tablePartition.intervalType == IntervalType.TIME_UNIT:
-                return TablePartitionConfig(
-                    partitionField=entity.tablePartition.columns[0]
+        if (
+            hasattr(entity, "tableProfilerConfig")
+            and hasattr(entity.tableProfilerConfig, "partitioning")
+            and entity.tableProfilerConfig.partitioning
+        ):
+            return entity.tableProfilerConfig.partitioning
+
+        if (
+            hasattr(entity, "serviceType")
+            and entity.serviceType == DatabaseServiceType.BigQuery
```

Comment:
do we have a followup issue to generalise this? Instead of handling it only for BQ cc @ulixius9

Reply:
what do you mean? The partitioning config will be handled for all type of entities. We need that check as some BQ table will require partitioning (whether it is set or not in the config).

Reply:
sorry I think I got confused, forgot about the special BQ cols. I said nothing